File: engineering/codebase/smartqr.frontend-services/src/domain/codes/core/enums/BarcodeFormat.ts
/** Defines the symbology a code is rendered as (mirrors backend `barcodeFormat`). */
export const BarcodeFormat = {
  /** Refers to a 2D QR code. */
  QrCode: "qrCode",
  /** Refers to a 2D Data Matrix code. */
  DataMatrix: "dataMatrix",
  /** Refers to a 2D PDF417 stacked barcode. */
  Pdf417: "pdf417",
  /** Refers to a 2D Aztec code. */
  Aztec: "aztec",
  /** Refers to a 1D Code 128 barcode. */
  Code128: "code128",
  /** Refers to a 1D EAN-13 retail barcode. */
  Ean13: "ean13",
  /** Refers to a 1D UPC-A retail barcode. */
  UpcA: "upcA",
} as const;

export type BarcodeFormat = (typeof BarcodeFormat)[keyof typeof BarcodeFormat];

File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Core/Enums/BarcodeFormat.cs
namespace SmartQr.Domain.Codes.Core.Enums;

/// <summary>Defines the symbology used to render a code.</summary>
public enum BarcodeFormat
{
    /// <summary>Represents a QR code — 2D.</summary>
    QrCode,

    /// <summary>Represents a Data Matrix — 2D.</summary>
    DataMatrix,

    /// <summary>Represents a PDF417 — 2D stacked.</summary>
    Pdf417,

    /// <summary>Represents an Aztec code — 2D.</summary>
    Aztec,

    /// <summary>Represents a Code 128 — 1D.</summary>
    Code128,

    /// <summary>Represents an EAN-13 — 1D retail product barcode.</summary>
    Ean13,

    /// <summary>Represents a UPC-A — 1D retail product barcode.</summary>
    UpcA,
}
